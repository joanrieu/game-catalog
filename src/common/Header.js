import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";
import logoUrl from "../../assets/Video_game_controller_icon_designed_by_Maico_Amorim.svg";

const Wrapper = styled.header`
  background-color: #333333;
  color: white;
  display: flex;
  align-items: center;
`;

const Logo = styled.img`
  width: 32px;
  height: 32px;
  padding: 16px;
`;

const Title = styled.h1`
  font-weight: bold;
  font-size: 24px;
`;

export default function Header({ title }) {
  return (
    <Wrapper>
      <Link to="/">
        <Logo src={logoUrl} />
      </Link>
      <Title>{title}</Title>
    </Wrapper>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired
};
