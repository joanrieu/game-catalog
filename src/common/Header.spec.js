import React from "react";
import { render, cleanup } from "react-testing-library";
import Header from "./Header";

describe("Header", () => {
  afterEach(cleanup);

  it("has a logo and a title", () => {
    const cmp = render(<Header title="TITLE" />);
    expect(cmp.asFragment()).toMatchSnapshot();
  });
});
