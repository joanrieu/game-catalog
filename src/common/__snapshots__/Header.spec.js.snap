// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has a logo and a title 1`] = `
<DocumentFragment>
  <header
    class="sc-bdVaJa DnvYQ"
  >
    <a
      href="/"
    >
      <img
        alt="Home"
        class="sc-bwzfXH ZQnxd"
        src="ASSET"
      />
    </a>
    <h1
      class="sc-htpNat OaCfN"
    >
      TITLE
    </h1>
  </header>
</DocumentFragment>
`;
